kernel/Kconfig.idle

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2025 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+menu "Kernel idle options"
+
+config IDLE_ENTER_HOOK
+	bool "Execute idle enter hook"
+	help
+	  Execute idle_exit_hook() when entering idle.
+
+config IDLE_DEFAULT_ROUTINE
+	bool "Execute idle default routine"
+	default y
```

Comment:
any options executed by the default routine should depend on this option, right? e.g. policy stuff

Reply:
well, optimistically if pm is enabled it would select the hook and implement a custom routine. Right now everything depends on the default, so it would practically need to be added everywhere, so I just leave it default to be disabled for now, until more stuff implements the hook than not :)

Reply:
I foresee this will be a source of confusion, people being able to enable options that do nothing is a bad idea.
Why don't we accept the PM subsystem is garbage and propose a new PM subsystem that evolves in parallel, as an experimental thing that will eventually replace what we have?

Reply:
PM_NG or PM2? maybe AM? Naming things is hard. I like the parallel pm subsystem idea.

Reply:
> I foresee this will be a source of confusion, people being able to enable options that do nothing is a bad idea. Why don't we accept the PM subsystem is garbage and propose a new PM subsystem that evolves in parallel, as an experimental thing that will eventually replace what we have?

@gmarull I really want to remind you that calling something "garbage" is just outright disrespectful to the people and community members who worked on it. If I'm not mistaken neither you and I are native speakers, but I believe we both know there are *many* other words in the English language to qualify something that could use some improvement/refactoring/re-engineering, or even a complete rewrite, than "garbage".

Reply:
I think it's an accurate word in this case. I'm open to use any more polite synonyms.

Reply:
To be clear, the solution in this PR is not a replacement of PM, its an "overlay"/ higher level abstraction which matches more than the system power state, we still need PM for systems with complex power states, but those states can be abstracted to "just" an event latency, which matches this PM_EVTC in this PR :)